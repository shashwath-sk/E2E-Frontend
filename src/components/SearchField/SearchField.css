.filter-container{
    display: flex;
    flex-direction: row;
    /* justify-content: ; */
    align-items: center;
}

.filter-container >input{
    width: 270px;
    font-size: 18px;
    padding: 5px;
    background-color: inherit;
    color: rgb(165,165,165);
    border-style: hidden;
}