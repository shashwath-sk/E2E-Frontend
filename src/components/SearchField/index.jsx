import * as React from 'react';
import search1 from '../../assets/search/search1.png';
import PropTypes from 'prop-types';
import './SearchField.css';

export default function SearchField( { placeholder } ) {
    return(
        <div className='filter-container'>
            <input type="text" placeholder={placeholder} />
            <img src={search1} alt="search" />
        </div>
    );

}


SearchField.propTypes = {
    placeholder: PropTypes.string,
};