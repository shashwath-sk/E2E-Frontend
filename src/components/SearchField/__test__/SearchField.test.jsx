import SearchField from '..';
import React from 'react';
import { render, screen } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';

describe('SearchField', () => {
    it('should render SearchField', () => {
        const { asFragment } = render(
            <BrowserRouter>
                <SearchField />
            </BrowserRouter>
        );
        expect(asFragment).toMatchSnapshot();
    });
    
    it('should show search field with Collection types as placeholder when rendered', () => {
        render(
            <BrowserRouter>
                <SearchField placeholder="COLLECTION TYPES"/>
            </BrowserRouter>
        );
        expect(screen.getByPlaceholderText('COLLECTION TYPES')).toBeInTheDocument();
    });

});