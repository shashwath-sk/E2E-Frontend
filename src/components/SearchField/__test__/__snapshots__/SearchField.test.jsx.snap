// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchField should render SearchField 1`] = `[Function]`;
